internal/service/distribution/distribution.go

```diff
@@ -0,0 +1,172 @@
+	if ac != nil {
+		creds, err = getAuthCredsFunc(refDomain, s.client, *ac)
+		if err != nil {
+			return nil, err
+		}
+	}
+	resolver, _, err := s.nctlImageSvc.GetDockerResolver(ctx, refDomain, creds)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize remotes resolver: %s", err)
+	}
+
+	_, desc, err := resolver.Resolve(ctx, namedRef)
+	if err != nil {
+		// translate error definitions from containerd
+		switch {
+		case cerrdefs.IsNotFound(err):
+			return nil, errdefs.NewNotFound(err)
+		case errors.Is(err, dockerresolver.ErrInvalidAuthorization):
+			return nil, errdefs.NewUnauthenticated(err)
+		default:
+			// this error is thrown when auth fails
+			var errStatus cremoteerrors.ErrUnexpectedStatus
+			if errors.As(err, &errStatus) {
+				return nil, errdefs.NewUnauthenticated(err)
+			}
+			return nil, err
+		}
+	}
+
+	fetcher, err := resolver.Fetcher(ctx, namedRef)
+	if err != nil {
+		return nil, err
+	}
+
+	rc, err := fetcher.Fetch(ctx, desc)
+	if err != nil {
+		switch {
+		case cerrdefs.IsNotFound(err):
+			return nil, errdefs.NewNotFound(err)
+		default:
+			return nil, err
+		}
+	}
+
+	res, err := io.ReadAll(rc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read fetch result: %w", err)
+	}
+
+	if dgst := desc.Digest.Algorithm().FromBytes(res); dgst != desc.Digest {
+		return nil, fmt.Errorf("digest mismatch: %s != %s", dgst, desc.Digest)
+	}
+
+	var platforms []ocispec.Platform
+	switch {
+	case desc.MediaType == ocispec.MediaTypeImageManifest ||
+		desc.MediaType == containerdimages.MediaTypeDockerSchema2Manifest:
+		var manifest ocispec.Manifest
+		if err := json.Unmarshal(res, &manifest); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal manifest: %w", err)
+		}
+
+		// fetch the image to get the platform
+		rc, err := fetcher.Fetch(ctx, manifest.Config)
+		if err != nil {
+			switch {
+			case cerrdefs.IsNotFound(err):
+				return nil, errdefs.NewNotFound(err)
+			default:
+				return nil, err
+			}
+		}
+
+		imageRes, err := io.ReadAll(rc)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read image: %w", err)
+		}
+
+		if dgst := manifest.Config.Digest.Algorithm().FromBytes(imageRes); dgst != manifest.Config.Digest {
+			return nil, fmt.Errorf("image digest mismatch: %s != %s", dgst, manifest.Config.Digest)
+		}
+
+		var image ocispec.Image
+		if err := json.Unmarshal(imageRes, &image); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal image: %w", err)
+		}
+
+		platforms = []ocispec.Platform{image.Platform}
+	case desc.MediaType == ocispec.MediaTypeImageIndex ||
+		desc.MediaType == containerdimages.MediaTypeDockerSchema2ManifestList:
+		var index ocispec.Index
+		if err := json.Unmarshal(res, &index); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal index: %w", err)
+		}
+		for _, manifest := range index.Manifests {
+			platforms = append(platforms, *manifest.Platform)
+		}
+	}
+
+	return &registrytypes.DistributionInspect{
+		Descriptor: ocispec.Descriptor{
```

Comment:
FWIW, Docker only returns `MediaType`, `Digest`, and `Size` in the descriptor. Should this really return so much extra info?

Reply:
Yeah, you are correct, good catch.

https://github.com/moby/moby/blob/b5d5fef7aa68f160b817edef186c9b18ab996f4f/api/server/router/distribution/distribution_routes.go#L93C1-L97C4

The spec website also documents the same limited parameters. I'll make a new revision without sending all the extra Descriptor fields.

Reply:
Fields are removed in my latest revision, PTAL